### PR TITLE
fix(jobs, reports): fail report as failed immediately on job creation failure

### PR DESCRIPTION
**Pull request by syncush**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                       |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |

Further  information:
@NivLipetz @enudler any idea how to create an integration test for that, in integration I can't retrieve the report_id, I was thinking about using `GET /last_reports`, wdyt?

### CODE REVIEW

File: src/jobs/models/jobManager.js [hunk 1]
```diff
@@ -65,18 +65,21 @@ module.exports.createJob = async (job) => {
...
         if (job.cron_expression) {
             addCron(jobId, insertedJob, job.cron_expression, configData);
         }
         logger.info('Job deployed successfully');
-        return createResponse(jobId, job, reportId);
+        return createResponse(jobId, job, report.report_id);
     } catch (error) {
         logger.error(error, 'Error occurred trying to create new job');
+        if (report) {
+            logger.info('Removing the failed job report');
```
Comment (enudler) [inline src/jobs/models/jobManager.js:80]: logger.info(`Changing report: ${report.report_id} to failed after it failed to start`);

File: src/jobs/models/jobManager.js [hunk 2]
```diff
@@ -276,18 +279,23 @@ async function createJobRequest(jobId, reportId, jobBody, dockerImage, configDat
...
     const scheduledJob = new CronJob(cronExpression, async function () {
+        let report = null;
         try {
             if (job.enabled === false) {
                 logger.info(`Skipping job with id: ${jobId} as it's currently disabled`);
             } else {
                 const latestDockerImage = await dockerHubConnector.getMostRecentRunnerTag();
                 const test = await testsManager.getTest(job.test_id);
-                const report = await createReportForJob(test, job);
+                report = await createReportForJob(test, job);
```
Comment (enudler) [inline src/jobs/models/jobManager.js:289]: Is it possible to extract these functions to be common?

**Comment by enudler** [comment]
@syncush 

> @NivLipetz @enudler any idea how to create an integration test for that, in integration I can't retrieve the report_id, I was thinking about using GET /last_reports, wdyt?

I would go with /tests/test_id/reports/

File: src/jobs/models/jobManager.js (hunk 2 above)
Comment (syncush) [reply src/jobs/models/jobManager.js:289]: I do agree we need to extract the logic to a function

**Pull request merged**